src/Strider.c

```diff
@@ -0,0 +1,198 @@
+  if(s_step_average<1) return;
+
+  const GRect inset = grect_inset(layer_get_bounds(layer), GEdgeInsets(2));
+  graphics_context_set_fill_color(ctx, GColorYellow);
+
+  int trigangle = DEG_TO_TRIGANGLE(360 * s_step_average / s_step_goal);
+  // draw a very narrow radial (it's just a line)
+  graphics_fill_radial(ctx, inset, GOvalScaleModeFitCircle, 12,
+    trigangle - 1000,
+    trigangle);
+}
+
+static void window_load(Window *window) {
+  GRect window_bounds = layer_get_bounds(s_window_layer);
+
+  // Dots for the progress indicator
+  s_dots_layer = layer_create(window_bounds);
+  layer_set_update_proc(s_dots_layer, dots_layer_update_proc);
+  layer_add_child(s_window_layer, s_dots_layer);
+
+  // Progress indicator
+  s_progress_layer = layer_create(window_bounds);
+  layer_set_update_proc(s_progress_layer, progress_layer_update_proc);
+  layer_add_child(s_window_layer, s_progress_layer);
+
+  // Average indicator
+  s_average_layer = layer_create(window_bounds);
+  layer_set_update_proc(s_average_layer, average_layer_update_proc);
+  layer_add_child(s_window_layer, s_average_layer);
+
+  // Create a layer to hold the current time
+  s_time_layer = text_layer_create(
+      GRect(0, PBL_IF_ROUND_ELSE(82, 78), window_bounds.size.w, 38));
+  text_layer_set_text_color(s_time_layer, GColorWhite);
+  text_layer_set_background_color(s_time_layer, GColorClear);
+  text_layer_set_font(s_time_layer,
+                      fonts_get_system_font(FONT_KEY_BITHAM_30_BLACK));
+  text_layer_set_text_alignment(s_time_layer, GTextAlignmentCenter);
+  layer_add_child(s_window_layer, text_layer_get_layer(s_time_layer));
+
+  // Create a layer to hold the current step count
+  s_step_layer = text_layer_create(
+      GRect(0, PBL_IF_ROUND_ELSE(58, 54), window_bounds.size.w, 38));
+  text_layer_set_background_color(s_step_layer, GColorClear);
+  text_layer_set_font(s_step_layer,
+                      fonts_get_system_font(FONT_KEY_GOTHIC_24_BOLD));
+  text_layer_set_text_alignment(s_step_layer, GTextAlignmentCenter);
+  layer_add_child(s_window_layer, text_layer_get_layer(s_step_layer));
+
+  // Subscribe to health events if we can
+  if(is_health_available()) {
+    health_service_events_subscribe(health_handler, NULL);
+  }
+}
+
+static void window_unload(Window *window) {
+  layer_destroy(text_layer_get_layer(s_time_layer));
+  layer_destroy(text_layer_get_layer(s_step_layer));
+  layer_destroy(s_dots_layer);
+  layer_destroy(s_progress_layer);
+  layer_destroy(s_average_layer);
+}
+
+void init() {
+  s_window = window_create();
+  s_window_layer = window_get_root_layer(s_window);
+  window_set_background_color(s_window, GColorBlack);
+
+  window_set_window_handlers(s_window,
+                             (WindowHandlers) {
+                                 .load = window_load,
+                                 .unload = window_unload
+                             });
+
+  ColorLoser = GColorPictonBlue;
```

Comment:
Move these assignments to the top of `init()` to homogenize the window creation, and since they have no dependent statements before them.